server/src/test/java/org/elasticsearch/search/fetch/subphase/FieldFetcherTests.java

```diff
@@ -1002,6 +1002,33 @@ public void testLastFormatWins() throws IOException {
         assertThat(fields.get("date_field").getValues().get(1), equalTo("12"));
     }
 
+    public void testNestedPrefix() throws IOException {
+        XContentBuilder mapping = XContentFactory.jsonBuilder()
+            .startObject()
+            .startObject("_doc")
+            .startObject("properties")
+            .startObject("foo")
+            .field("type", "nested")
+            .startObject("properties")
+            .startObject("nestedField")
+            .field("type", "keyword")
+            .endObject()
+            .endObject()
+            .endObject()
+            .startObject("foo_bar")
+            .field("type", "double")
+            .endObject()
+            .endObject()
+            .endObject()
+            .endObject();
```

Comment:
nit: with checkstyle rules these large XContentBuilder setups are hard to read. I'm wondering if we can use the new Java text blocks to create the mapping from a readable String representation, but then again I haven't found a quick example after some searching just now. Happy to leave it as is for now but if you have an idea how to remedy this I'd be glad to see and copy going forward.

Reply:
Changed to use Java text blocks. Yes, it makes it easier to read.